mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -251,106 +258,230 @@ static void populateDefaults() {
-  block->push_back(outputHostAllocOp);
+  // Emit verification logic.
+  // Create a new block
+  Block *verifyResultsBlock = verifyFuncOp.addEntryBlock();
+
+  // %c0 = constant 0: index
+  auto c0IndexOp = builder.create<ConstantIndexOp>(builder.getUnknownLoc(), 0);
+  verifyResultsBlock->push_back(c0IndexOp);
+
+  // %result = alloca() : memref<1xi32>
+  SmallVector<int64_t, 1> oneElementVector({1});
+  auto resultMemRefType = MemRefType::get(
+      ArrayRef<int64_t>(oneElementVector.begin(), oneElementVector.end()),
+      builder.getIntegerType(32));
+  auto cmpResultAllocOp =
+      builder.create<AllocaOp>(builder.getUnknownLoc(), resultMemRefType);
+  verifyResultsBlock->push_back(cmpResultAllocOp);
+
+  // %c0_i32 = constant 0 : i32
+  // %c1_i32 = constant 1 : i32
+  auto c0ConstantInt32Op = builder.create<ConstantIntOp>(
+      builder.getUnknownLoc(), 0, builder.getIntegerType(32));
+  verifyResultsBlock->push_back(c0ConstantInt32Op);
+  auto c1ConstantInt32Op = builder.create<ConstantIntOp>(
+      builder.getUnknownLoc(), 1, builder.getIntegerType(32));
+  verifyResultsBlock->push_back(c1ConstantInt32Op);
+
+  // store %c1_i32, %result[%c0] : memref<1xi32>
+  auto storeOp1 =
+      builder.create<StoreOp>(builder.getUnknownLoc(), c1ConstantInt32Op,
+                              cmpResultAllocOp, ValueRange{c0IndexOp});
+  verifyResultsBlock->push_back(storeOp1);
+
+  // %%c1 = constant 1 : index
+  auto c1IndexOp = builder.create<ConstantIndexOp>(builder.getUnknownLoc(), 1);
+  verifyResultsBlock->push_back(c1IndexOp);
+
+  // Emit constant index Ops for loop upper bounds
+  auto indexOp0 = builder.create<ConstantIndexOp>(builder.getUnknownLoc(),
+                                                  outputDimension[0]);
+  verifyResultsBlock->push_back(indexOp0);
+  auto indexOp1 = builder.create<ConstantIndexOp>(builder.getUnknownLoc(),
+                                                  outputDimension[1]);
+  verifyResultsBlock->push_back(indexOp1);
+  auto indexOp2 = builder.create<ConstantIndexOp>(builder.getUnknownLoc(),
+                                                  outputDimension[2]);
+  verifyResultsBlock->push_back(indexOp2);
+  auto indexOp3 = builder.create<ConstantIndexOp>(builder.getUnknownLoc(),
+                                                  outputDimension[3]);
+  verifyResultsBlock->push_back(indexOp3);
+
+  // scf.for %arg0 = %c0 to %c128 step %c1 {
+  //  scf.for %arg1 = %c0 to %c30 step %c1 {
+  //    scf.for %arg2 = %c0 to %c30 step %c1 {
+  //      scf.for %arg3 = %c0 to %c128 step %c1 {
+  //        %cpu_result = load %CpuResults[%arg0, %arg1, %arg2, %arg3] :
+  //        memref<128x30x30x128xf32> %gpu_result = load %GpuRsults[%arg0,
+  //        %arg1, %arg2, %arg3] : memref<128x30x30x128xf32> %cmp_result = cmpf
+  //        "oeq", %cpu_result, %gpu_result : f32
+  //
+  //        scf.if %cmp_result {
+  //        } else {
+  //          store %c0_i32, %result[%c0] : memref<1xi32>
+  //        }
```

Comment:
Can you use scf.if !%cmp_result {}?

Reply:
I would think so. This is what the prototype does.